# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/DeleteNodeTransaction.h
# Commit: 876b1031a6ae
# Full Hash: 876b1031a6ae41a87ed146976f279b96bde9c7f6
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-03 21:40:10
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 1: Make each transaction class grab their members with local variable before touching the DOM tree r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69302
# ==============================================================================

diff -r 7f069649d6cf -r 876b1031a6ae editor/libeditor/DeleteNodeTransaction.h
--- a/editor/libeditor/DeleteNodeTransaction.h	Fri Apr 03 06:03:25 2020 +0300
+++ b/editor/libeditor/DeleteNodeTransaction.h	Fri Apr 03 01:38:22 2020 +0000
@@ -23,7 +23,7 @@
  */
 class DeleteNodeTransaction final : public EditTransactionBase {
  protected:
-  DeleteNodeTransaction(EditorBase& aEditorBase, nsINode& aNodeToDelete);
+  DeleteNodeTransaction(EditorBase& aEditorBase, nsIContent& aContentToDelete);
 
  public:
   /**
@@ -31,10 +31,10 @@
    * it cannot remove the node from its parent.
    *
    * @param aEditorBase         The editor.
-   * @param aNodeToDelete       The node to be removed from the DOM tree.
+   * @param aContentToDelete    The node to be removed from the DOM tree.
    */
   static already_AddRefed<DeleteNodeTransaction> MaybeCreate(
-      EditorBase& aEditorBase, nsINode& aNodeToDelete);
+      EditorBase& aEditorBase, nsIContent& aContentToDelete);
 
   /**
    * CanDoIt() returns true if there are enough members and can modify the
@@ -57,13 +57,13 @@
   RefPtr<EditorBase> mEditorBase;
 
   // The element to delete.
-  nsCOMPtr<nsINode> mNodeToDelete;
+  nsCOMPtr<nsIContent> mContentToDelete;
 
   // Parent of node to delete.
   nsCOMPtr<nsINode> mParentNode;
 
   // Next sibling to remember for undo/redo purposes.
-  nsCOMPtr<nsIContent> mRefNode;
+  nsCOMPtr<nsIContent> mRefContent;
 };
 
 }  // namespace mozilla